Detectors/FIT/FT0/workflow/src/ft0-flp-workflow.cxx

```diff
@@ -71,27 +81,36 @@ WorkflowSpec defineDataProcessing(ConfigContext const& configcontext)
   auto isExtendedMode = configcontext.options().get<bool>("tcm-extended-mode");
   auto disableRootOut = configcontext.options().get<bool>("disable-root-output");
   auto askSTFDist = !configcontext.options().get<bool>("ignore-dist-stf");
+  auto isNewDecoder = configcontext.options().get<bool>("new-decoder");
   o2::conf::ConfigurableParam::updateFromString(configcontext.options().get<std::string>("configKeyValues"));
   LOG(info) << "WorkflowSpec FLPWorkflow";
-  //Type aliases
-  //using RawReaderFT0trgInput = o2::fit::RawReaderFIT<o2::ft0::RawReaderFT0BaseNorm,true>;
+  // Type aliases
+  // using RawReaderFT0trgInput = o2::fit::RawReaderFIT<o2::ft0::RawReaderFT0BaseNorm,true>;
   using RawReaderFT0 = o2::fit::RawReaderFIT<o2::ft0::RawReaderFT0BaseNorm, false>;
-  //using RawReaderFT0trgInputExt = o2::fit::RawReaderFIT<o2::ft0::RawReaderFT0BaseExt,true>;
+  // using RawReaderFT0trgInputExt = o2::fit::RawReaderFIT<o2::ft0::RawReaderFT0BaseExt,true>;
   using RawReaderFT0ext = o2::fit::RawReaderFIT<o2::ft0::RawReaderFT0BaseExt, false>;
   using MCLabelCont = o2::dataformats::MCTruthContainer<o2::ft0::MCLabel>;
   o2::header::DataOrigin dataOrigin = o2::header::gDataOriginFT0;
   //
   WorkflowSpec specs;
-  if (isExtendedMode) {
-    specs.emplace_back(o2::fit::getFITDataReaderDPLSpec(RawReaderFT0ext{dataOrigin, dumpReader}, askSTFDist));
-    if (!disableRootOut) {
-      specs.emplace_back(o2::fit::FITDigitWriterSpecHelper<RawReaderFT0ext, MCLabelCont>::getFITDigitWriterSpec(false, false, dataOrigin));
+  if (!isNewDecoder) {
+    if (isExtendedMode) {
+      specs.emplace_back(o2::fit::getFITDataReaderDPLSpec(RawReaderFT0ext{dataOrigin, dumpReader}, askSTFDist));
+      if (!disableRootOut) {
+        specs.emplace_back(o2::fit::FITDigitWriterSpecHelper<RawReaderFT0ext, MCLabelCont>::getFITDigitWriterSpec(false, false, dataOrigin));
+      }
+    } else {
+      specs.emplace_back(o2::fit::getFITDataReaderDPLSpec(RawReaderFT0{dataOrigin, dumpReader}, askSTFDist));
+      if (!disableRootOut) {
+        specs.emplace_back(o2::fit::FITDigitWriterSpecHelper<RawReaderFT0, MCLabelCont>::getFITDigitWriterSpec(false, false, dataOrigin));
+      }
     }
   } else {
-    specs.emplace_back(o2::fit::getFITDataReaderDPLSpec(RawReaderFT0{dataOrigin, dumpReader}, askSTFDist));
-    if (!disableRootOut) {
-      specs.emplace_back(o2::fit::FITDigitWriterSpecHelper<RawReaderFT0, MCLabelCont>::getFITDigitWriterSpec(false, false, dataOrigin));
-    }
+#if defined(__has_include)
```

Comment:
perhaps you should not offer the option `new-decoder` or at least print some error message when you are not in the correct compiler environment?

Reply:
> Is the output supposed to be the same as with the old decoder?
> Particularly, does it work with the current CTF version?

Yes, output is the same as from old decoder, except Trigger inputs. Unfortunately I didn't check it with current CTF version yet. But I checked with other devices(Digit writer), and this decoder has sorted by BC-orbit IDs digits within output container. And actually I think it will not work correctly with current CTF version because ChIDs is unsorted(we discussed it). In parallel I'm preparing and testing such PR for CTFCoder.

> perhaps you should not offer the option `new-decoder` or at least print some error message when you are not in the correct compiler environment?

Yes, I'm planning to implement this in next patch(+ some other kind of protections within new decoder, in TODO list). But first we would like to make stress tests for this decoder at the FLP next week. Just to be sure that it can handle with highest rates in real life(I tested it only by using emulated readout.exe->StfB->DPL->StfS chain w/o ECS at FLP machine)